g3/git/git.py

```diff
@@ -1,10 +1,18 @@
-from typing import Optional
+from typing import List, Optional
 
 from pydantic import BaseModel
 
+from g3.git.diff import Diff, get_filenames, get_files_changed
 from g3.git.shell import Shell
 
 
+class ModelData(BaseModel):
+    filenames: List[str]
+    diffs: List[Diff]
+    repo_name: str
+    branch_name: str
```

Comment:
Why don't we move all these properties within `GitInfo` so that we can access all git information with a single instantiation (i.e. GitInfo()`?

Reply:
Done!

Reply:
Just to remind that now the changes will be computed on initialisation of class instead by explicitly calling the function